sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -651,13 +652,31 @@ class SparkSession private(
     withActive {
       val plan = tracker.measurePhase(QueryPlanningTracker.PARSING) {
         val parsedPlan = sessionState.sqlParser.parsePlan(sqlText)
-        if (args.nonEmpty) {
-          PosParameterizedQuery(parsedPlan, args.map(lit(_).expr).toImmutableArraySeq)
-        } else {
-          parsedPlan
+        parsedPlan match {
+          case compoundBody: CompoundBody => compoundBody
```

Comment:
shall we fail here if `args` is nonEmpty?

Reply:
We still have that check in case of "standard" query. For `CompoundBody`, I am not sure if we need this check. @dbatomic what do you think?

Reply:
OK I see it, but the code isn't very readable. I'd prefer more explicit way
```
case compoundBody: CompoundBody =>
  if has args fails
  execute script
case p =>
  if has args PosParameterizedQuery... else ...
```

Reply:
Resolved.